src/main/java/org/example/services/ItemProducer.java

```diff
@@ -0,0 +1,41 @@
+package org.example.services;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.concurrent.BlockingQueue;
+import org.example.database.DatabaseInterface;
+import org.example.model.Item;
+
+public class ItemProducer implements Runnable {
+  private final BlockingQueue<Item> itemQueue;
+  private final DatabaseInterface dbConnection;
+
+  public ItemProducer(final BlockingQueue<Item> itemQueue, final DatabaseInterface dbConnection) {
+    this.itemQueue = itemQueue;
+    this.dbConnection = dbConnection;
+  }
+
+  @Override
+  public void run() {
+    try {
+      final ResultSet resultSet = dbConnection.getItemData();
+      while (resultSet.next()) {
+        final int id = resultSet.getInt("id");
```

Comment:
can you use a java class rather than using Resultset and extracting individual columns (optional)